DependencyInjection/Compiler/EntityListenerPass.php

```diff
@@ -52,7 +52,12 @@ public function process(ContainerBuilder $container)
                     $this->attachToListener($container, $name, $id, $attributes);
                 }
 
-                $container->getDefinition($resolver)->addMethodCall('register', array(new Reference($id)));
+                if (isset($attributes['lazy']) && $attributes['lazy']) {
+                    $listener = $container->getDefinition($id);
+                    $container->getDefinition($resolver)->addMethodCall('registerService', array($listener->getClass(), $id));
```

Comment:
Good point. It could be added to the interface, however I'm not sure if this would count as a BC break (classes implementing this outside of this bundle would have to be updated). 

We could also check for the existence of the method and throw an exception if the configured resolver does not. It's up to the maintainers, I think.